scipy/stats/_multivariate.py

```diff
@@ -5066,3 +5067,161 @@ def rvs(self, size=1, random_state=None):
+        If `seed` is an int, a new ``RandomState`` instance is used, seeded
+        with seed.
+        If `seed` is already a ``RandomState`` or ``Generator`` instance,
+        then that object is used.
+        Default is `None`.
+
+    Notes
+    -----
+    This distribution generates unit vectors uniformly distributed on
+    the surface of a sphere. These can be interpreted as random directions.
+
+    For example, if `dim=3`, 3D vectors from the surface of math:`S^2`
+    will be sampled.
+
+    References
+    ----------
+    .. [1] Marsaglia, G. (1972). "Choosing a Point from the Surface of a
+           Sphere". Annals of Mathematical Statistics. 43 (2): 645-646.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.stats import random_direction
+    >>> x = random_direction.rvs(3)
+    >>> np.linalg.norm(x)
+    1.
+
+    This generates one random direction, a vector on the surface of
+    :math:`S^2`.
+
+    Alternatively, the object may be called (as a function) to fix the `dim`
+    parameter, return a "frozen" unitary_group random variable:
+
+    >>> rv = random_direction(5)
+
+    """
+
+    def __init__(self, seed=None):
+        super().__init__(seed)
+        self.__doc__ = doccer.docformat(self.__doc__)
+
+    def __call__(self, dim=None, seed=None):
+        """Create a frozen n-dimensional uniform direction distribution.
+
+        See `random_direction` for more information.
+        """
+        return random_direction_frozen(dim, seed=seed)
+
+    def _process_parameters(self, dim):
+        """Dimension N must be specified; it cannot be inferred."""
+        if dim is None or not np.isscalar(dim) or dim < 1 or dim != int(dim):
+            raise ValueError("Dimension of rotation must be specified,"
+                             "and must be a scalar greater than 0.")
+
+        return dim
+
+    def rvs(self, dim, size=1, random_state=None):
+        """Draw random samples from S(N-1).
+
+        Parameters
+        ----------
+        dim : integer
+            Dimension of space (N).
+        size : int or tuple of ints, optional
+            Given a shape of, for example, (m,n,k), m*n*k samples are
+            generated, and packed in an m-by-n-by-k arrangement.
+            Because each sample is N-dimensional, the output shape
+            is (m,n,k,N). If no shape is specified, a single (N-D)
+            sample is returned.
+
+        Returns
+        -------
+        rvs : ndarray
+            Random direction vectors
+
+        """
+        random_state = self._get_random_state(random_state)
+
+        error_message_size = "`size` must be a tuple of integers or an int."
+        if isinstance(size, int):
+            size = (size, )
+        elif isinstance(size, tuple):
+            if not all(isinstance(v, int) for v in size):
+                raise ValueError(error_message_size)
+        else:
+            raise ValueError(error_message_size)
```

Comment:
This might be enough to get at what you really need.
```suggestion
        size = np.atleast_1d(size)
        if size.ndim != 1 or not np.issubdtype(size, np.integer):
            raise raise ValueError(error_message_size)
```
And if you wanted to more completely validate the input, you would want to check that the values are non-negative.

On the other hand, this is almost immediately passed into `standard_normal` after concatenating `dim`, so nothing other than wrapping a scalar into an iterable of some sort is really needed; `multivariate_normal` will give an error message that is pretty easy to debug with. Going either way - stripping this down and letting `standard_normal` take care of the validation or going all the way with the validation - is fine with me, but I'd suggest choosing one extreme or the other.